java/src/org/openqa/selenium/chrome/ChromeDriverFinder.java

```diff
@@ -0,0 +1,45 @@
+package org.openqa.selenium.chrome;
+
+import org.openqa.selenium.internal.Require;
+import org.openqa.selenium.manager.SeleniumManager;
+import org.openqa.selenium.os.ExecutableFinder;
+
+import java.io.File;
+import java.util.logging.Logger;
+
+public class ChromeDriverFinder {
```

Comment:
Can't we not create a generic driver finder, which accepts the variables as args (binary name, JVM argument, error message details) and all the browser specific finders just calls into that generic driver finder ? That way we could avoid duplicating the logic across all the driver finders.

Reply:
+1 We could abstract DriverFinder

Reply:
Yes, we can definitely optimize this solution if we think it's the right approach.